src/python/pants/backend/python/rules/resolve_requirements.py

```diff
@@ -0,0 +1,89 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from future.utils import text_type
+
+from pants.backend.python.subsystems.python_native_code import PexBuildEnvironment
+from pants.backend.python.subsystems.python_setup import PythonSetup
+from pants.engine.fs import Digest, Snapshot, UrlToFetch
+from pants.engine.isolated_process import ExecuteProcessRequest, ExecuteProcessResult
+from pants.engine.rules import optionable_rule, rule
+from pants.engine.selectors import Get
+from pants.util.objects import datatype, hashable_string_list, string_optional, string_type
+from pants.util.strutil import create_path_env_var
+
+
+class ResolveRequirementsRequest(datatype([
+  ('requirements', hashable_string_list),
+  ('output_filename', string_type),
+  ('entry_point', string_optional),
+  ('interpreter_constraints', hashable_string_list),
+])):
+  pass
+
+
+class ResolvedRequirementsPex(datatype([
+  ('directory_digest', Digest),
+  ('requirements', hashable_string_list)
+])):
+  pass
```

Comment:
I'm not sure what would be useful to have in this `datatype`. 

For example, should we include `stdout` and `stderr`, or is that irrelevant because if the `ExecuteProcessRequest` fails then we will already print those out for the user?

Is it repetitive to have `requirements` again, since it is already in the `ResolveRequirementsRequest`? Should we include `entry_point` and `interpreter_constraints`?

Reply:
> For example, should we include stdout and stderr, or is that irrelevant because if the ExecuteProcessRequest fails then we will already print those out for the user?

Yep, that's my feeling.

> Should we include entry_point and interpreter_constraints?

No, I don't think so.

Reply:
Okay, so to clarify, this sounds good to be `directory_digest` + `requirements`? I think that makes a lot of sense.

Reply:
I think we should remove `requirements` - the caller should already know them if it needs them.

Reply:
Yea, agreed. Sorry, was initially reviewing from phone.